src/bits/bit_array.rs

```diff
@@ -0,0 +1,181 @@
+use super::{BitArray, BooleanOps};
+use bitvec::prelude::{BitArr, Lsb0};
+use std::{
+    fmt::Debug,
+    ops::{BitAnd, BitAndAssign, BitOr, BitOrAssign, BitXor, BitXorAssign, Index, Not},
+};
+
+/// Bit store size of `XorSecretShare` in bytes.
+const BIT_STORE_SIZE_IN_BYTES: usize = 4;
+type U8_4 = BitArr!(for BIT_STORE_SIZE_IN_BYTES * 8, in u8, Lsb0);
```

Comment:
Using `u8` blocks as backing data type makes converting from a byte array into BitArray easier. If we ever need a backing data type that is a multiple of 8 (i.e. 40), it's also easy to just change `BIT_STORE_SIZE_IN_BYTES` rather than using `u32`. Not sure if there's a performance overhead using four blocks of `u8` instead of one block of `u32`. We can measure the difference later if we decide that 32-bit is enough for xor secret shares.

Reply:
I think that you will find that we'll need a different backing at some point.  A lot of the operations we're talking about here work more neatly if this is backed by u32 or u64 (or even usize, as much as Alex hates that).

Reply:
Should we go ahead and implement `BitArray64` backed by u64?

Reply:
Changed it to `BitArray64`. I kept the backing type as `u8` because that plays nicely with `serialize`/`deserialize` for `Message` trait, converting to/from a byte slice. With `u32` or `u64`, we need to call extra `to_le_bytes` and `from_le_bytes` on each communication.